components/brave_wallet_ui/common/slices/endpoints/pricing.endpoints.ts

```diff
@@ -96,80 +91,37 @@ export const pricingEndpoints = ({
             data: { assetRatioService },
           } = baseQuery(undefined)
 
-          if (!ids.length) {
-            throw new Error('no token ids provided for price lookup')
+          if (!requests.length) {
+            throw new Error('no requests provided for price lookup')
           }
 
-          // dedupe ids to prevent duplicate price requests
-          const uniqueIds = [...new Set(ids)]
-            // skip flagged coins such as testnet coins and null/undefined ids
-            .filter((id) => id && id !== SKIP_PRICE_LOOKUP_COINGECKO_ID)
-
-          const chunkedParams = []
-          for (let i = 0; i < uniqueIds.length; i += maxBatchSizePrice) {
-            chunkedParams.push(uniqueIds.slice(i, i + maxBatchSizePrice))
+          const chunkedRequests = []
+          for (let i = 0; i < requests.length; i += maxBatchSizePrice) {
+            chunkedRequests.push(requests.slice(i, i + maxBatchSizePrice))
           }
 
           // Use maxConcurrentPriceRequests concurrent HTTP requests to
           // fetch prices, in batch of maxBatchSizePrice.
           const results = await mapLimit(
-            chunkedParams,
+            chunkedRequests,
             maxConcurrentPriceRequests,
-            async function (params: string[]) {
+            async function (requests: BraveWallet.AssetPriceRequest[]) {
               const { success, values } = await assetRatioService.getPrice(
-                params,
-                [toCurrency],
-                timeframe ?? BraveWallet.AssetPriceTimeframe.Live,
+                requests,
+                vsCurrency,
               )
 
               if (success && values) {
```

Comment:
Do we still need batching? 
`AssetRatioService::GetPrice` does one http request. What is the limit of its json size?

Reply:
Removing batching logic as discussed on Slack: [3a750df](https://github.com/brave/brave-core/pull/31054/commits/3a750dfb9a8d403967358730b4d9db8c8f2da612)